internal/gatewayapi/route.go

```diff
@@ -751,10 +788,11 @@ func (t *Translator) processHTTPRouteParentRefListener(route RouteContext, route
 					IsHTTP2:               routeRoute.IsHTTP2,
 					SessionPersistence:    routeRoute.SessionPersistence,
 					Timeout:               routeRoute.Timeout,
+					Retry:                 routeRoute.Retry,
 				}
-				if routeRoute.Traffic != nil {
+				if retry := routeRoute.GetRetry(); retry != nil {
```

Comment:
@guydc @liorokman do you remember why this code is needed ? it doesnt feel like it should be here

Reply:
cc @shawnh2 seems like this was introduced in https://github.com/envoyproxy/gateway/commit/6f6d8689d18f8b2011eaaf62c39ccf6930476e3e but the logic doesnt hold true because you seperate out route timeout and retries from BTP timeout and retries

Reply:
my vote would be to delete this block L800-803

Reply:
removed, let's see how CI reacted.